### PR TITLE
mgr/dashboard/telemetry: reduce telemetry dashboard preview size

### PR DESCRIPTION
**Pull request by ljflores**
Keeping perf channel metrics on the telemetry preview form may cause
laggy loading times and slow behavior in the text box if the report
is particularly long. As a workaround, we can remove performance metrics
from the preview form and notify the users that they may still download
a full report upon selecting the "Download" button.

For the future, it might be good to implement this so telemetry sends
a compressed report to the server.

Fixes: https://tracker.ceph.com/issues/53831
Signed-off-by: Laura Flores <lflores@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ljflores** [comment]
Preliminary teuthology run:
http://pulpito.front.sepia.ceph.com/lflores-2022-01-12_16:17:06-rados-wip-telemetry-dashboard-distro-default-smithi/

**Review by yaarith — APPROVED** [review]
looks like a good workaround

**Comment by yaarith** [comment]
let's also have an approval from the dashboard team

**Review by pereman2 — APPROVED** [review]
worked locally

**Comment by ljflores** [comment]
> worked locally

Thanks @pereman2! @neha-ojha if the teuthology tests I ran are sufficient, I believe this can be merged. Thoughts?

**Comment by neha-ojha** [comment]
> > worked locally
> 
> Thanks @pereman2! @neha-ojha if the teuthology tests I ran are sufficient, I believe this can be merged. Thoughts?

looks sufficient to me, I'll let the dashboard team have the final say

**Comment by pereman2** [comment]
> @pereman2 I know you already approved this, but I just want to be sure before merging this: Are the teuthology tests that were scheduled sufficient for this PR?
> 
> http://pulpito.front.sepia.ceph.com/lflores-2022-01-12_16:17:06-rados-wip-telemetry-dashboard-distro-default-smithi/

@ljflores They look good. We mostly rely on the dashboards tests to verify dashboard related behaviour so, I would say if you are willing to write tests for this, we all would be less worried about something breaking.

**Comment by ljflores** [comment]
@pereman2 good point. I added a unit test.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts:111]: This seems to be the recommended way to perform deep-copy with JS objects, but when a reviewer stumbles upon this line they'll likely scratch their heads... Why not using Lodash [`_.cloneDeep()`](https://lodash.com/docs/4.17.15#cloneDeep)?

**Review by epuertat — APPROVED** [review]
LGTM!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
Comment (ljflores) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts:111]: Thanks for telling me about this, I'll use it in the future. And thanks again so much for taking a look at this!

**Pull request merged**
